platform/default/src/mbgl/storage/offline_database.cpp

```diff
@@ -791,7 +809,7 @@ std::exception_ptr OfflineDatabase::deleteRegion(OfflineRegion&& region) try {
 
     evict(0);
     assert(db);
-    db->exec("PRAGMA incremental_vacuum");
+    db->exec("VACUUM");
```

Comment:
I don't know enough here, but is this something that should be scheduled for some idle time in the future?

Reply:
This method has become heavier, right? is it expected and should it be also mentioned in the method description?

Reply:
Also is it an option to schedule `VACUUM` separately, so that it is performed in a different callchain?

Reply:
It has not become heavier, I could not measure any performance difference between calling `VACUUM` and `PRAGMA incremental_vacuum` in this case. The reason why I'm replacing with `VACUUM` is because `PRAGMA incremental_vacuum` was not working and we would need to migrate existing databases to a new version to make it work.

Reply:
@julianrex I think that is a good idea, but we would have to keep a state on the database that it is pending a `VACUUM`, because the app could be killed before we run it. Another problem I found when working on this code is that we only request data from the network after we query the database synchronously. One thing that could be done already is make the network request before accessing the cache to increase parallelism, but ideally access to the database should also be asynchronous and not block the network thread.